src/clientSideScene/sceneEntities.ts

```diff
@@ -510,32 +435,184 @@ export class SceneEntities {
   ) => {
     await kclManager.updateAst(modifiedAst, false)
     await this.tearDownSketch({ removeAxis: false })
-    this.setupSketch({ sketchPathToNode, forward, up, position: origin })
+    await this.setupSketch({
+      sketchPathToNode,
+      forward,
+      up,
+      position: origin,
+      maybeModdedAst: kclManager.ast,
+    })
+    this.setupSketchIdleCallbacks(sketchPathToNode)
   }
-  setUpDraftArc = async (
+  setUpDraftSegment = async (
     sketchPathToNode: PathToNode,
     forward: [number, number, number],
-    up: [number, number, number]
+    up: [number, number, number],
+    origin: [number, number, number],
+    segmentName: 'line' | 'tangentialArcTo' = 'line',
+    shouldTearDown = true
   ) => {
-    await this.tearDownSketch({ removeAxis: false })
-    await new Promise((resolve) => setTimeout(resolve, 100))
-    this.setupSketch({
+    const _ast = JSON.parse(JSON.stringify(kclManager.ast))
+
+    const variableDeclarationName =
+      getNodeFromPath<VariableDeclaration>(
+        _ast,
+        sketchPathToNode || [],
+        'VariableDeclaration'
+      )?.node?.declarations?.[0]?.id?.name || ''
+    const sg = kclManager.programMemory.root[
+      variableDeclarationName
+    ] as SketchGroup
+    const lastSeg = sg.value.slice(-1)[0] || sg.start
+
+    const index = sg.value.length // because we've added a new segment that's not in the memory yet, no need for `-1`
+
+    let modifiedAst = addNewSketchLn({
+      node: kclManager.ast,
+      programMemory: kclManager.programMemory,
+      to: [lastSeg.to[0], lastSeg.to[1]],
+      from: [lastSeg.to[0], lastSeg.to[1]],
+      fnName: segmentName,
+      pathToNode: sketchPathToNode,
+    }).modifiedAst
+    modifiedAst = parse(recast(modifiedAst))
+
+    const draftExpressionsIndices = { start: index, end: index }
+
+    if (shouldTearDown) await this.tearDownSketch({ removeAxis: false })
+    const { truncatedAst, programMemoryOverride, sketchGroup } =
+      await this.setupSketch({
+        sketchPathToNode,
+        forward,
+        up,
+        position: origin,
+        maybeModdedAst: modifiedAst,
+        draftExpressionsIndices,
+      })
+    this.setupDraftSegmentCallbacks({
       sketchPathToNode,
-      draftSegment: 'tangentialArcTo',
+      sketchGroup,
       forward,
       up,
+      origin,
+      segmentName,
+      truncatedAst,
+      programMemoryOverride,
+      variableDeclarationName,
```

Comment:
Lastly we add some callbacks, here I'm using another method `setupDraftSegmentCallbacks` because it get used in a few places, but you can add you own directly with
```typescript
sceneInfra.setCallbacks({
  onMove: async (args) => {
    // ...
  },
  // ...
}
```
You might need more than just `onMove`, but `onMove` is going to be the most important because that's what you'll use to update the numbers in your ast to get it to animate.

Reply:
side note edit, I just realised `setupDraftSegmentCallbacks` is not used more than once, so I might move it out of the method.